packages/context/src/context-view.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugFactory from 'debug';
+import {promisify} from 'util';
+import {Binding} from './binding';
+import {BindingFilter} from './binding-filter';
+import {Context} from './context';
+import {
+  ContextEventListener,
+  ContextEventType,
+  Subscription,
+} from './context-listener';
+import {Getter} from './inject';
+import {ResolutionSession} from './resolution-session';
+import {resolveList} from './value-promise';
+const debug = debugFactory('loopback:context:view');
+
+/**
+ * `ContextView` provides a view for a given context chain to maintain a live
+ * list of matching bindings and their resolved values within the context
+ * hierarchy.
+ *
+ * This class is the key utility to implement dynamic extensions for extension
+ * points. For example, the RestServer can react to `controller` bindings even
+ * they are added/removed/updated after the application starts.
+ *
+ */
+export class ContextView<T = unknown> implements ContextEventListener {
+  protected _cachedBindings: Readonly<Binding<T>>[] | undefined;
+  protected _cachedValues: T[] | undefined;
+  private _subscription: Subscription | undefined;
+
+  constructor(
+    protected readonly ctx: Context,
```

Comment:
_re-posting an older comment_

Let's not use abbreviations please and call the property `context`.